src/Symfony/Component/ErrorRenderer/ErrorRenderer/JsonErrorRenderer.php

```diff
@@ -40,12 +40,18 @@ public function render(FlattenException $exception): string
     {
         $debug = $this->debug && ($exception->getHeaders()['X-Debug'] ?? true);
 
+        if ($debug) {
+            $message = $exception->getMessage();
+        } else {
+            $message = 404 === $exception->getStatusCode() ? 'Sorry, the page you are looking for could not be found.' : 'Whoops, looks like something went wrong.';
```

Comment:
Sorry but I'm totally👎 on this. The message does not make any sense in an API context.

Reply:
Would you like a generic message like: `Not Found (404 error)` or do you have any other message in mind?

Reply:
I prefer no message at all. What we put here it's arbitrary.

Reply:
Other suggestion for this 404 message is this one:
```
The server has not found anything matching the Request-URI
```
Taken from https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

Reply:
This is how the message would look for JSON format:
```json
{
    "title": "Not Found",
    "status": 404,
    "detail": "The server has not found anything matching the Request-URI"

}
```

Reply:
> Whoops, looks like something went wrong.

Why does it say "something" went wrong when it's clear what went "wrong" based on the status code?
Why does it say "wrong" when in theory I can return a successful error code from `getStatusCode`?

Reply:
> > Whoops, looks like something went wrong.

> Why does it say "something" went wrong when it's clear what went "wrong" based on the status code?

Because I want to be more friendly with an explicit and verbose error message. It's not arbitrary, it was taken from here https://github.com/symfony/symfony/blob/2.0/src/Symfony/Component/HttpKernel/Debug/ExceptionHandler.php#L77-L83 (since Symfony 2.0)

> Why does it say "wrong" when in theory I can return a successful error code from getStatusCode?

Which wouldn't make sense to me.

Reply:
The title already gives the text explanation of the status code. Anything else is just arbitrary and I don't see why you keep insisting on the detail property.

Reply:
Please, revert these changes if you still think it's useless and arbitrary, but then I'll be agaisnt to have "detail" property for debug mode only, because (1) it is not intended for debugging information according to rfc; (2) for the same request consistently we must return the same response, at least this would apply to the members defined in the rfc, "detail" is one of them.

Anything else is considered an "extensions". In this case "exception" property (rendered on debug mode only) is an extension property.

It's problable that we're using the wrong names here and we should have called them "message" & "trace" respectively as extensions of the problem detail for debug mode only, so we can get rid of them on non-debug mode.